### ISSUE TITLE
Idea's thread

### ISSUE THREAD

**Issue by jhjacobs81** [linuxkettle/AppShop#6]
to prevent us from too many issues, lets throw up balls here!

idealy, the client would run a daemon on the OS that takes care of "the environment" like installing python for python based programs, Ruby for Ruby based programs, etc. 

also, the programmers would follow some basic guidelines when it comes down to where the app gets installed (app bundles). preferably in /Applications and/or ~/Applications this makes it easy to backup your whole application stuff? 

would be nice to have some visual guidelines as well for people who start programming. We could help them make apps look uniform and sexy? 

Privacys is a thing. everything we have should be encrypted and not be readable by human eyes (save for statistics) so we dont have to wory about data breaches etc. maybe work with certificates? 

it should be possible to export your shop config, for users without an acount. on a new computer they could simply import it, and the system takes care of restoring the apps. combine this with a simple backup of the app locations, and i think its pretty fool proof?

Also, it should be possible to use an account. in the long run we would want to feature paid apps, and users should not pay more then once for each app. we would need nothing other then an email/password combination.

**Comment by kettle-7** [comment]
> to prevent us from too many issues,

We're at issue #6 already!

> idealy, the client would run a daemon on the OS that takes care of "the environment" like installing python for python based programs, Ruby for Ruby based programs, etc.

That could be done without a daemon. But one could be useful for automatic updates.

> also, the programmers would follow some basic guidelines when it comes down to where the app gets installed (app bundles). preferably in /Applications and/or ~/Applications this makes it easy to backup your whole application stuff?

So long as they follow some standard expectations (in other words, write "clean" code) they'll be fine. It's when you hardcode paths that portability becomes an issue.

> would be nice to have some visual guidelines as well for people who start programming. We could help them make apps look uniform and sexy?

This one is a little harder, because you want apps to look native. Qt does this well.

> Privacys is a thing. everything we have should be encrypted and not be readable by human eyes (save for statistics) so we dont have to wory about data breaches etc. maybe work with certificates?

I make desktop apps, I have never dealt with this stuff before. Basically, all should be anonymous and, honestly, we shouldn't be collecting much data other than "Here's how many people have downloaded your app" and "Here's what version everybody uses".

> it should be possible to export your shop config, for users without an acount. on a new computer they could simply import it, and the system takes care of restoring the apps. combine this with a simple backup of the app locations, and i think its pretty fool proof?

I was just thinking about a ~/Settings folder where you point `XDG_CONFIG_HOME` to. It would make this easier.

> Also, it should be possible to use an account. in the long run we would want to feature paid apps, and users should not pay more then once for each app. we would need nothing other then an email/password combination.

This is definitely way outside of what I thought I was doing when making an "App Shop". How about we walk one step at a time? I'm sure I'll get there, but I'm not mentally prepared for so big a project!

**Comment by jhjacobs81** [comment]
thats why this is the idea's thread ;-) 

so, /Application for systemwide, ~/Application for personal apps (would also not require root rights, maybe even better) and ~/Settings for app settings. sounds good to me :) 

The daemon indeed would be for updates etc. this would help people be notified of new app updates without having to manually check. People are dumb, lazy creatures by nature ;-) we invented computers for that :) 

Like i said.. im a visionair.. Aim for the stars, reach the sky ;-)

**Comment by kettle-7** [comment]
> The daemon indeed would be for updates etc. this would help people be notified of new app updates without having to manually check. People are dumb, lazy creatures by nature ;-) we invented computers for that :)

I know people who've used Windows for years and don't know what a file extension is.

I think an Automatic Update feature would need to be able to be turned off in settings somewhere. I think we'd need some degree of separation between app updates and system updates -- on many Linux distributions this line is blurred. macOS and Windows have proprietary update systems, FreeBSD is a mess, Linux varies by distro, I wonder if we even need system update integration...

**Comment by jhjacobs81** [comment]
System updates would not be my first choice no.. lets stick to app updates first ;-)

**Comment by kettle-7** [comment]
Sounds good.

**Comment by jhjacobs81** [comment]
its a bit silent? :)

**Comment by ghost** [comment]
Could use Appimage.github.io as a source for apps

As well.

**Comment by kettle-7** [comment]
Yeah, I started adding the ability to add your own repository to the app catalogue, like fLatpak can.

**Comment by ghost** [comment]
Oh cool, didn't know that. Will there be a default one?

**Comment by kettle-7** [comment]
http://raw.githubusercontent.com/linuxkettle/aPpShOp/main/apps/ls

Yes.

**Comment by ghost** [comment]
Which languages are inuse btw?

**Comment by kettle-7** [comment]
QML mostly, it's based on JavaScript. There's the odd C++ wrapper, but I don't need to even touch those once I've made them. One to talk to libcurl and the other to load the QML.

**Comment by ghost** [comment]
Would you be open to me helping after I learn it?

**Comment by kettle-7** [comment]
Of course, I will probably need all the help I can get!

If you have a fast processor then maybe you can compile some stuff.

**Comment by kettle-7** [comment]
Okay, I'm putting something out this week that'll be of interest to you, a framework for making graphical apps in Python based on Qt.

**Comment by kettle-7** [comment]
And I will completely remake AppShelf with it, I might need some help ;~)

**Comment by kettle-7** [comment]
> I'm putting something out this week

Check out <https://github.com/arbyon-osloos/platform>, I'm adding more to it all the time. It's stable now, you can make a multi-page app with text boxes, buttons, switches etc. and all in Python.
